common/src/main/java/com/wynntils/screens/OverlayManagementScreen.java

```diff
@@ -0,0 +1,651 @@
+                        poseStack,
+                        mouseX
+                                - RenderUtils.getToolTipWidth(
+                                        RenderUtils.componentToClientTooltipComponent(HELP_TOOLTIP_LINES),
+                                        FontRenderer.getInstance().getFont()),
+                        mouseY,
+                        100,
+                        HELP_TOOLTIP_LINES,
+                        FontRenderer.getInstance().getFont(),
+                        false);
+            }
+        }
+
+        super.render(poseStack, mouseX, mouseY, partialTick); // This renders widgets
+    }
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+        if (testMode) return super.mouseClicked(mouseX, mouseY, button);
+
+        // Order:
+        //  - Corners
+        //  - Edges
+        //  - OverlayArea
+
+        // reset
+        resetSelection();
+
+        Overlay overlay = selectedOverlay;
+
+        Vec2 mousePos = new Vec2((float) mouseX, (float) mouseY);
+
+        for (Map.Entry<Corner, Vec2> corner : overlay.getCornersMap().entrySet()) {
+            float distance = corner.getValue().distanceToSqr(mousePos);
+            if (distance < MAX_CLICK_DISTANCE) {
+                selectedCorner = corner.getKey();
+                selectionMode = SelectionMode.Corner;
+
+                return false;
+            }
+        }
+
+        for (Edge value : Edge.values()) {
+            float minX, maxX, minY, maxY;
+
+            switch (value) {
+                case Top -> {
+                    minX = overlay.getRenderX();
+                    maxX = overlay.getRenderX() + overlay.getWidth();
+                    minY = overlay.getRenderY() - MAX_CLICK_DISTANCE / 2f;
+                    maxY = overlay.getRenderY() + MAX_CLICK_DISTANCE / 2f;
+                }
+                case Left -> {
+                    minX = overlay.getRenderX() - MAX_CLICK_DISTANCE / 2f;
+                    maxX = overlay.getRenderX() + MAX_CLICK_DISTANCE / 2f;
+                    minY = overlay.getRenderY();
+                    maxY = overlay.getRenderY() + overlay.getHeight();
+                }
+                case Right -> {
+                    minX = overlay.getRenderX() + overlay.getWidth() - MAX_CLICK_DISTANCE / 2f;
+                    maxX = overlay.getRenderX() + overlay.getWidth() + MAX_CLICK_DISTANCE / 2f;
+                    minY = overlay.getRenderY();
+                    maxY = overlay.getRenderY() + overlay.getHeight();
+                }
+                case Bottom -> {
+                    minX = overlay.getRenderX();
+                    maxX = overlay.getRenderX() + overlay.getWidth();
+                    minY = overlay.getRenderY() + overlay.getHeight() - MAX_CLICK_DISTANCE / 2f;
+                    maxY = overlay.getRenderY() + overlay.getHeight() + MAX_CLICK_DISTANCE / 2f;
+                }
+                default -> {
```

Comment:
Is Edge an enum? I thought the switch expression barfed on `default` if all enum values were covered..?

Reply:
This prevents variable might not have been initialized.

Reply:
As a warning? That sounds really odd. How on earth could we get there with `value` uninitialized? Is that the IntelliJ compiler?

Reply:
![image](https://user-images.githubusercontent.com/49001742/181127498-377277c3-50c5-4099-a8b2-b11fa9f5ebbe.png)

Reply:
I had to go dig this down. It is by design, in http://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#13.4.26. The problem is that in theory, we can compile this class first, then add a new enum and just recompile the Edge class, and then this code would have an unaccounted-for path. So this is correct.